library/src/main/java/com/novoda/downloadmanager/BatchFileBuilder.java

```diff
@@ -0,0 +1,46 @@
+package com.novoda.downloadmanager;
+
+/**
+ * Adds {@link BatchFile} to the parent {@link Batch} using the {@link BatchBuilder}
+ * and {@link BatchFileBuilder} fluent APIs.
+ */
+public interface BatchFileBuilder {
+
+    /**
+     * Sets {@link BatchFileBuilder} to build a {@link BatchFile} with a corresponding
+     * {@param downloadFileId} that can be used to lookup a download file belonging to
+     * a given {@link Batch}.
+     *
+     * @param downloadFileId to flag a {@link BatchFile} with.
+     * @return {@link BatchFileBuilder}.
+     */
+    BatchFileBuilder withIdentifier(DownloadFileId downloadFileId);
+
+    /**
+     * Sets {@link BatchFileBuilder} to build a {@link BatchFile} that will be saved
+     * to the given {@param path}.
```

Comment:
Maybe specify here that the path is only referring to the directory that will contain the file.
And also, is it relative or absolute?